Uzupełnienia

rczajka opened the pull request:
Pozwoliłem sobie przegrzebać trochę i uzupełnić kod, nie jestem specem od FB2, więc możesz chcieć przejrzeć. Z regresji: przeniosłem motta z epigrafów do tekstu głównego.

rczajka committed:
smaller image for tests

rczajka committed:
fixes #2325: repetitions in editor list

rczajka committed:
Allow specyfing fallback values for BookInfo fields when creating a WLDocument.

rczajka committed:
added WLDocument.as_cover: treat covers more like ebooks

rczajka committed:
Better Unicode handling in errors.

rczajka committed:
pdf: respect inherited cover

rczajka committed:
fixes #2356: set User-Agent

rczajka committed:
Merge remote-tracking branch 'mgorny/fb2'

rczajka committed:
Many FB2 fixes

mgorny commented:
Postaram się to przejrzeć w ciągu tygodnia. Choć kodu tak wiele, że chyba tylko praktycznie wytestuję i ręcznie sprawdzę zgodność wyników ze specyfikacją.

mgorny commented:
Przetestowałem wstępnie na przykładowych plikach i za dobrze to nie wygląda.
1. `liryki-lozanskie-zal-rozrzutnika.xml` — wrzuca motywy do tekstu (przedtem wycinał je poprawnie):
   
   ```
   <v>Miłość, Przyjaźń, Serce, SamotnośćKochanek, druhów, ileż was spotkałem,</v>
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ```
2. `tako-rzecze-zaratustra.xml` — zdaje się, że wpada w jakąś pętelkę, której końca się nie doczekałem.

Zechcesz przyjrzeć się temu? Ja niestety chwilowo, że się tak z angielska wyrażę, jestem krótki na czasie.

mgorny commented:
Zaratustrę jednak w końcu skończył przerabiać. Coś długo mu to szło, i strasznie często powtarza niepotrzebnie `xmlns:dc="http://purl.org/dc/elements/1.1/"`, ale to kosmetyka i łatwo będzie obejść, w najgorszym przypadku. Pozostaje więc kwestia motywów.

rczajka commented:
Ok, biorę to na siebie.

rczajka committed:
Scalable metrics in Cover

rczajka committed:
Add WL logo on cover

rczajka committed:
Add table of themes directly to HTML.

rczajka committed:
forgotten image

rczajka committed:
Spaces in themes list in HTML.

rczajka committed:
Use Pillow-compatible imports.

rczajka committed:
Added theme table in test

rczajka committed:
fixes #2620: remove old tagline

rczajka committed:
Fixes #2631: losing whitespace in epub/mobi stanzas.

rczajka committed:
#2773: support info in pdf/epub/mobi.

rczajka committed:
PDF cover info fix

rczajka committed:
Add funders info.

rczajka committed:
funders in txt

rczajka committed:
Legal note fix.

rczajka committed:
typo

rczajka committed:
Add thank-you note in EPUB and PDF.

aflukasz committed:
Fixing thank-you note in epub

Commit:
Accept dates in like "2 poł. XIX w."

Commit:
Merge branch '__images'

Commit:
tests for handling "2 poł XIX w." dates + utf8 check

Commit:
Handle all date formats in a primitive way (always take the beginning of period)

Commit:
More on date format

rczajka committed:
Fix broken txt test.

Commit:
add genre to picture_info

rczajka committed:
Metadata can have lang attribute.

Commit:
fixes for dates etc

Commit:
merging master

Commit:
use lists not tuples since tuples break things

rczajka committed:
Covers with logos.

rczajka committed:
Use cover with logos by default.

rczajka committed:
Insert anchors before, not inside elements.

quiris11 committed:
Paragraphs formatting improvements

No margins between .paragraph classes plus text indentation between two adjacent .paragraph classes.

quiris11 committed:
Fix for incorrect .ttf MIME type

quiris11 committed:
Transition from square bracket to superscripted footnotes

quiris11 committed:
CSS style footnotes clean up

quiris11 committed:
[Mobi] CSS improvements for paragraph and footnotes

quiris11 committed:
[ePub] Fix for inconsistent styles between on-line reader and ePub.

quiris11 committed:
[Mobi] Fix for inconsistent styles between on-line reader and Mobi.

quiris11 committed:
[ePub Mobi] Removed unnecessary annotations clutter. Make them according typography rules

quiris11 committed:
[ePub, Mobi] Smaller text-indent in paragraphs.

quiris11 committed:
Use CSS styles also in HTML TOC

quiris11 committed:
[Mobi] Mobi output improvements

1. Produce dual Mobi file by default (Mobi7 + KF8). Much nicer and
richer experience on Kindle devices.
2. Embed fonts in Mobi.
3. Remove obsolete Mobi stylesheet (mobi.py uses epub css by default)
4. Add flag with-full-fonts to epub.py as it is required for mobi.py to
produce working subset of fonts by calibre.
5. Ignore margins option by default to ensure that result mobi file
will use Kindle devices own margins.

quiris11 committed:
Fix for Epubcheck Version 3.0.1 validation warning

WARNING: OEBPS/toc.ncx: meta@dtb:uid content 'sample' should conform to
unique-identifier in content.opf:
'http://wolnelektury.pl/katalog/lektura/sample'

quiris11 committed:
Fix for missing xml declaration of content.opf and toc.ncx

Lack of xml declaration causes failed ePub validation with Epubcheck
Version 3.0.1

quiris11 committed:
Fix for dc:creator and dc:date tag – make it as described in OPF 2.0.1 spec.

According to an example from OPF 2.0.1 spec:
<dc:creator opf:file-as="King, Martin Luther Jr." opf:role="aut">
        Rev. Dr. Martin Luther King Jr.
</dc:creator>

Regarding to dc:date change: Sigil throws  with xsi:type in dc:date.

quiris11 committed:
[ePub] Added conversion from 3-letter to 2-letter (if available) language code

The change-set is intended to fix various ebook readers problems
regarding not recognized 3-letter (bibliographic) language code.

quiris11 committed:
[ePub] Fix for error validation with epubcheck 3.0.1 in multi-volume books

ERROR: example.epub/OPS/toc.ncx(24,47): assertion failed: different
playOrder values for navPoint/navTarget/pageTarget that refer to same
target

quiris11 committed:
[epub, mobi] CSS improvements - better interoperability on various readers

quiris11 committed:
[epub, mobi] Fix for a problem with epubs, mobi in drama ebooks without acts

quiris11 committed:
Revert "[epub, mobi] Fix for a problem with epubs, mobi in drama ebooks without acts"

This reverts commit 0081ccb8e51026b5af824b3c641648d23155f1e1.

quiris11 committed:
Revert "Revert "[epub, mobi] Fix for a problem with epubs, mobi in drama ebooks without acts""

This reverts commit afc2b870c1646c8d6c1b754d4ee44aa059e834c9.

quiris11 committed:
Revert deb4aee..88cabfe

This rolls back to commit deb4aee6a2185645b197a9a8524e7ed2232a7410.

quiris11 committed:
[epub, mobi] Fix for a problem with epubs, mobi in drama ebooks without acts

Better aproach than previous one.

quiris11 committed:
[txt] return Windows friendly txt file with \r\n line ends instead of \n

POSIX editors recognize Windows line end. Notepad on Windows does not
recognize POSIX line ends.

quiris11 committed:
[epub, mobi] fix for notes without margin-bottom. Issue #2567

quiris11 committed:
[epub, mob] Paragraphs inside note should be justified

quiris11 committed:
[epub, mobi] motto_podpis impovements

quiris11 committed:
[epub, mobi] fix for hanging single letter conjunctions - currenty disabled

for future use.

quiris11 committed:
[epub, mobi] further nota tag rendering improvements

quiris11 committed:
[epub, mobi] didaskalia tag rendering improvement

quiris11 committed:
[epub, mobi] note tag styling improvement - one more time…

rczajka committed:
Add support for custom covers (coverBarColor, coverBoxPosition), remove the logo-in-white-box option.

rczajka committed:
Fix HTML test.

rczajka committed:
Fix HTML TOC themes issue.

rczajka committed:
Remove unused image.

rczajka committed:
Fix some errors in scripts.

rczajka committed:
Wide PDF cover.

rczajka committed:
Fixes #3031: Sponsor info in PDF.

rczajka committed:
Fixes #2570: Text spilling into fragments from outside.

quiris11 committed:
Merge recent changes from branch 'fnp/librarian/master'

Sync between fork and origin

quiris11 committed:
[epub, mobi] Paragraph indentation improvements plus support for initials in chapters

rczajka committed:
Fix escaping stuff in PDF.

rczajka committed:
Reorganizing partner code a little.

rczajka committed:
fix in book2partner

rczajka committed:
PDF cover: higher resolution, lower quality.

rczajka committed:
Merge remote-tracking branch 'quiris/master' into quiris

rczajka committed:
[TXT] Newline after last line, update test.

quiris11 committed:
[epub] fix for calibre warning - cover does not need linear=no attribute

Calibre said „There are items marked as non-linear in the <spine>.
These will be displayed in random order by different ebook readers.
Some will ignore the non-linear attribute, some will display them at
the end or the beginning of the book and some will fail to display them
at all. Instead of using non-linear items simply place the items in the
order you want them to be displayed.”

quiris11 committed:
[epub, mobi] fix for IT #3356 - new line char is problemtic in toc.ncx - replace it with space

quiris11 committed:
Merge remote-tracking branch 'upstream/master'

Sync between librarian master and fork

quiris11 committed:
Merge remote-tracking branch 'upstream/master'

rczajka committed:
String comparison bug.

quiris11 committed:
Merge remote-tracking branch 'upstream/master'

quiris11 committed:
[epub, mobi] Add early hyphenate support for powiesc, opowiadania and wywiad. TODO: language detection

quiris11 committed:
[epub, mobi] hyphenator - added language detection and more hyph dictionaries

quiris11 committed:
[epub, mobi] hyphenate all kinds of documents and annotation style improvement

quiris11 committed:
fix impovement for IT #3356 - workaround rare problem

rczajka committed:
[HTML] fixes #3029: <motto> contains blocks.

quiris11 committed:
[epub, mobi] encode in utf-8 instead of entities. Result: smaller files :)

quiris11 committed:
[epub, mobi] - xsl output impovements  required for better annotation visual rendering on Kindle Paperwhite

quiris11 committed:
[epub] output improvements - clean xhtml instead of html tag soup

quiris11 committed:
[epub, mobi] - tabs clean up plus media query amzn-kf8 helpful during kindlegen conversion

quiris11 committed:
[epub, mobi]  book-text id – alignment with a rest of html output files

quiris11 committed:
[epub, mobi] added optional command line switch helpful with kindlegen conversion

quiris11 committed:
[epub, mobi] new switch use-kindlegen and html-toc switch improvement

quiris11 committed:
[epub] removed flag with-full-fonts as it is not required any more

quiris11 committed:
[mobi] not needed cover lines removed - epub intermediate file provides required cover; added support for second optional conversion tool - kindlegen

quiris11 committed:
Fix for not working --verbose optional switch

quiris11 committed:
Merge remote-tracking branch 'upstream/master'

merge upstream with my own fork

quiris11 committed:
[mobi] pep8

quiris11 committed:
[epub] pep8 tabs removed

quiris11 committed:
[epub] pep8

rczajka committed:
Fix font-optimizer for Perl 5.18.

rczajka committed:
Merge remote-tracking branch 'quiris/master'

rczajka committed:
Minor fixes.

rczajka committed:
Added developmentStage meta field.

rczajka committed:
Allow no editors info in pdf, epub.

rczajka committed:
Change extract_annotations return value.

rczajka committed:
Disable test for unsupported behaviour in pictures.

rczajka committed:
Fix in extract_annotations

rczajka committed:
Fix test.

rczajka committed:
Data for #3396: picture style, full creation date.

rczajka committed:
html.extract_annotations: Allow multiple footnote qualifiers. Use only ones accepted by editors.

rczajka committed:
Preliminary math and tables support.

rczajka committed:
Minor fixups.

rczajka committed:
FB2 footnotes fix.

rczajka committed:
Minor fixes.

rczajka committed:
Added support for <abstrakt>

janek37 committed:
pep8 and other code-style changes

janek37 committed:
epub fix for multiple authors (#3655)

janek37 committed:
minor fix in error reporting

janek37 committed:
removed dependency from wolnelektury

janek37 committed:
fix #3663: changed and updated sort keys

janek37 committed:
turn off hyphenator

janek37 committed:
fix conjunctions when not hyphenating

janek37 committed:
images in PDF

janek37 committed:
images in HTML

janek37 committed:
images in epub/mobi

janek37 committed:
epub: images in content.opf

janek37 committed:
minor fix in epub

better ids for images in content.opf

quiris11 committed:
set 5pt page margin instead of oversized 2em book margin

janek37 committed:
Merge pull request #8 from quiris11/master

set 5pt page margin instead of oversized 2em book margin

Commit:
fix for generating ebooks without pictures

Commit:
remove "import *"

Commit:
vertical space before subchapter

Commit:
fix custom pdf without wl font

janek37 committed:
footnote types in ebooks

janek37 committed:
suppress xmlns:... attributes in epub

janek37 committed:
verse/paragraph numbers in PDF (like in HTML)

janek37 committed:
temporarily disable line numbers (No room for new \count)
